Subject: Add AArch64 support to xserver-xorg
Author: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>

lnx_video.c parts are not existing in 1.14 branch.

Xserver was checked in AArch64 fastmodel (commercial one with graphics support).

http://patchwork.freedesktop.org/patch/12785/

Upstream-status: Pending

---
 hw/xfree86/os-support/linux/lnx_video.c |    4 ++--
 include/servermd.h                      |    7 +++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

--- xorg-server-1.13.1.orig/hw/xfree86/os-support/linux/lnx_video.c
+++ xorg-server-1.13.1/hw/xfree86/os-support/linux/lnx_video.c
@@ -509,11 +509,11 @@ xf86EnableIO(void)
             return FALSE;
         }
 #endif
     }
     close(fd);
-#elif !defined(__mc68000__) && !defined(__sparc__) && !defined(__mips__) && !defined(__sh__) && !defined(__hppa__) && !defined(__s390__) && !defined(__arm__) && !defined(__m32r__) && !defined(__nds32__)
+#elif !defined(__mc68000__) && !defined(__sparc__) && !defined(__mips__) && !defined(__sh__) && !defined(__hppa__) && !defined(__s390__) && !defined(__arm__) && !defined(__m32r__) && !defined(__nds32__) && !defined(__aarch64__)
     if (ioperm(0, 1024, 1) || iopl(3)) {
         if (errno == ENODEV)
             ErrorF("xf86EnableIOPorts: no I/O ports found\n");
         else
             FatalError("xf86EnableIOPorts: failed to set IOPL"
@@ -538,11 +538,11 @@ xf86DisableIO(void)
     if (!ExtendedEnabled)
         return;
 #if defined(__powerpc__)
     munmap(ioBase, 0x20000);
     ioBase = NULL;
-#elif !defined(__mc68000__) && !defined(__sparc__) && !defined(__mips__) && !defined(__sh__) && !defined(__hppa__) && !defined(__arm__) && !defined(__s390__) && !defined(__m32r__) && !defined(__nds32__)
+#elif !defined(__mc68000__) && !defined(__sparc__) && !defined(__mips__) && !defined(__sh__) && !defined(__hppa__) && !defined(__arm__) && !defined(__s390__) && !defined(__m32r__) && !defined(__nds32__) && !defined(__aarch64__)
     iopl(0);
     ioperm(0, 1024, 0);
 #endif
     ExtendedEnabled = FALSE;
 
--- xorg-server-1.13.1.orig/include/servermd.h
+++ xorg-server-1.13.1/include/servermd.h
@@ -243,10 +243,17 @@ SOFTWARE.
 #define BITMAP_BIT_ORDER       LSBFirst
 #define GLYPHPADBYTES		4
 /* ???? */
 #endif                          /* AMD64 */
 
+#if defined(__aarch64__) || defined(aarch64) || defined(__aarch64)
+#define IMAGE_BYTE_ORDER	MSBFirst
+#define BITMAP_BIT_ORDER       MSBFirst
+#define GLYPHPADBYTES		4
+/* ???? */
+#endif                          /* AArch64 */
+
 #if	defined(SVR4) && (defined(__i386__) || defined(__i386) ) ||	\
 	defined(__alpha__) || defined(__alpha) || \
 	defined(__i386__) || \
 	defined(__s390x__) || defined(__s390__)
 
